### PR TITLE
*: clean up code, remove BufStore in SessionVar.WriteStmtBufs

### PR DESCRIPTION
**Pull request by tiancaiamao**
As we reuse membuffer in TxnState, BufStore in WriteStmtBufs is unnecessary.

@winkyao @coocood

### CODE REVIEW

File: executor/write.go [hunk 1]
```diff
@@ -915,19 +911,14 @@ func (e *InsertExec) exec(goCtx goctx.Context, rows [][]types.Datum) (Row, error
...
 		if len(e.OnDuplicate) == 0 && !e.IgnoreErr {
 			txn.SetOption(kv.PresumeKeyNotExists, nil)
 		}
 		h, err := e.Table.AddRecord(e.ctx, row, false)
 		txn.DelOption(kv.PresumeKeyNotExists)
 		if err == nil {
-			if !sessVars.ImportingData {
-				e.ctx.StmtAddDirtyTableOP(DirtyTableAddRow, e.Table.Meta().ID, h, row)
-			}
+			e.ctx.StmtAddDirtyTableOP(DirtyTableAddRow, e.Table.Meta().ID, h, row)
```
Comment (tiancaiamao) [inline executor/write.go]: I think we should write binlog, whether or not it's importing data.

Comment (coocood) [reply executor/write.go]: It has some cost when merging dirty table.

Comment (tiancaiamao) [reply executor/write.go]: ok, `begin; stmt1; stmt2 ...` stmt2 can't see stmt1's write when it's importing data.
I'll revert here.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
This change would cause some potential problems:
Without that buffer, we can't guarantee add value and and index to be atomic.

The old logic:

```
add index to buf
if no error {
   add value to txn
   save buf to txn
}
```

Importing Data is unsafe in this case:

```
create table t (c auto increment, a int unique, b int unique)
insert into t values (null, 1, 0)
insert into t values (null, 0, 1)
replace into t values (null, 1, 1)
```

**Pull request closed without merging**